group4/81966_Georgi_Gerginov/Set/comp_implementation/student/student.cpp

```diff
@@ -0,0 +1,19 @@
+#include "student.h"
+
+Student::Student(const unsigned int& facnum) : m_facnum{facnum} {}
+
+const short int Student::compareTo(Comparable* other) const {
+    Student* pt_to_stu{dynamic_cast<Student*>(other)};
+
+    if(m_facnum < pt_to_stu -> m_facnum) {
+        return -1;
+    } else if(m_facnum == pt_to_stu -> m_facnum) {
+        return 0;
+    }
+
+    return 1;
```

Comment:
This would be a little bit better as else statement part of the if to show more explicitly that's the third option of the same if.